Mail/Views/AlertView.swift

```diff
@@ -47,23 +61,69 @@ struct CustomAlertModifier<AlertContent>: ViewModifier where AlertContent: View
 
     func body(content: Content) -> some View {
         content
-            .overlay {
-                Group {
-                    if isPresented {
-                        AlertView {
-                            alertView
-                        }
-                    }
+            .fullScreenCover(isPresented: $isPresented) {
+                AlertView {
+                    alertView
+                }
+            }
+            .onChange(of: isPresented) { _ in
+                // Disable the default slide over animation when presenting / dismissing the ViewController
+                UIView.setAnimationsEnabled(false)
+            }
+    }
+}
+
+struct CustomAlertItemModifier<Item, AlertContent>: ViewModifier where Item: Identifiable, AlertContent: View {
+    @Binding var item: Item?
+    let alertView: (Item) -> AlertContent
+
+    func body(content: Content) -> some View {
+        content
+            .fullScreenCover(item: $item) { item in
+                AlertView {
+                    alertView(item)
                 }
-                .animation(.default, value: isPresented)
+            }
+            .onChange(of: item?.id) { _ in
+                UIView.setAnimationsEnabled(false)
             }
     }
 }
 
+private struct ClearFullScreenView: UIViewRepresentable {
+    private class BackgroundRemovalView: UIView {
+        override func didMoveToWindow() {
+            super.didMoveToWindow()
+            clearBackgroundSuperviews(view: self)
+        }
+
+        private func clearBackgroundSuperviews(view: UIView, level: Int = 0) {
+            guard level < 5 else {
+                return
+            }
+
+            if let superview = view.superview {
+                superview.backgroundColor = .clear
+                clearBackgroundSuperviews(view: superview, level: level + 1)
+            }
+        }
+    }
+
+    func makeUIView(context: Context) -> UIView {
+        return BackgroundRemovalView()
```

Comment:
You are more into returning directly than I might be, but what is the prefered approach with implicit returns in closures ? I like to ommit retrun when possible.